cunumeric/eager.py

```diff
@@ -1622,3 +1622,19 @@ def unpackbits(
             self.array[:] = np.unpackbits(
                 src.array, axis=axis, bitorder=bitorder
             )
+
+    def _wrap(self, src: Any, new_len: int) -> None:
+        self.check_eager_args(src)
+        if self.deferred is not None:
+            self.deferred._wrap(src, new_len)
+        else:
+            src.array = np.ravel(src.array)
+            if src.size == new_len:
+                self.array[:] = src.array[:]
+            elif src.array.size > new_len:
+                self.array[:] = src.array[:new_len]
+            else:
+                out = src
+                while out.array.size < new_len:
+                    out.array = np.concatenate((out.array, src.array), axis=0)
+                self.array[:] = out.array[0:new_len]
```

Comment:
This can be simplified to something like:
```
reps = (new_len + local_arr.size - 1) // local_arr.size
if reps > 1:
    local_arr = local_arr.tile(reps)
self.array[:] = local_arr[:new_len]
```

Reply:
thank you